### ISSUE TITLE
Code signing on Windows fails when space in path

### ISSUE THREAD

**Issue by eriedl** [electron-userland/electron-builder#816]
- **Version**: 7.11.4

- **Target**: Windows (on Windows)

Building and signing my app on Windows for Windows results in the following error:

```
Signing My App.exe (certificate file "C:\Users\eriedl\AppData\Local\Temp\electron-builder-qZd8Ot\t-d18-0.p12")
  electron-builder Executing C:\Users\eriedl\.cache\winCodeSign\winCodeSign-1.4.2\windows-10\signtool.exe sign /t http://timestamp.verisign.com/scripts/timstamp.dll /f C:\Users\eriedl\AppData\Local\Temp\electron-builder-qZd8Ot\t-d18-0.p12 /d My App /du https://www.mycompany.io /p "password" C:\Users\eriedl\Documents\sources\ape\out\packages\win-unpacked\My App.exe +391ms
Unhandled rejection Error: Exit code: 1. Command failed: C:\Users\eriedl\.cache\winCodeSign\winCodeSign-1.4.2\windows-10\signtool.exe sign /t http://timestamp.verisign.com/scripts/timstamp.dll /f C:\Users\eriedl\AppData\Local\Temp\electron-builder-qZd8Ot\t-d18-0.p12 /d My App /du https://www.mycompany.io /p "password" C:\Users\eriedl\Documents\sources\ape\out\packages\win-unpacked\My App.exe
SignTool Error: An error occurred while attempting to load the signing
        certificate from: C:\Users\eriedl\AppData\Local\Temp\electron-builder-qZd8Ot\t-d18-0.p12
```

When I try to manually sign the executable, I also get an error. However, as soon as I surround all argument values that contain a space with double quotes, the signing works just fine:

```
C:\Users\eriedl\.cache\winCodeSign\winCodeSign-1.4.2\windows-10\signtool.exe sign /t http://timestamp.verisign.com/scripts/timstamp.dll /f resources\cert-store-authenticode.p12 /d "My App" /du https://www.mycompany.io /p "password" out\packages\win-unpacked\My App.exe
Done Adding Additional Store
Successfully signed: out\packages\win-unpacked\My App.exe
```

It might be a good idea to surround all paths with double quotes including the temporary path to `signtool.exe`, the p12 file, description option and executable.

**Comment by develar** [comment]
Issue not in the space (printed as not escaped, but executed correctly, I am sure). Strange.

**Comment by develar** [comment]
How do you specify path to certificate — file link, remote file or...?

**Comment by eriedl** [comment]
Well, this was another pebcack :( I had enclosed the `CSC_..` variable values with double quotes. That threw of the code signer off when trying to copy the p12 file to the temp directory. However, there was no real error as it created a small file in the temp directory. I copied it away after creation and discovered then that it was empty, which gave me the hint that the code signer can't find the file.
Maybe `downloadCertificate` can throw an error if the file specified by `file://` cannot be found?

Anyway, thanks again for your help!
